### PR TITLE
Fix glean tests

### PR DESCRIPTION
**Pull request by Dexterp37**
This is based on #2158, since I wasn't able to add commits to that one.

### Pull Request checklist

- [ ] This PR includes thorough **tests** or an explanation of why it does not
- [ ] This PR includes a **CHANGELOG entry** or does not need one
- [ ] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/test/java/org/mozilla/tv/firefox/TestFirefoxApplication.kt [hunk 1]
```diff
@@ -17,4 +20,12 @@ class TestFirefoxApplication : FirefoxApplication() {
      */
     override fun getSystemUserAgent(): String =
             "Mozilla/5.0 (Linux; Android 7.1.2) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Focus/2.2.0.2 Chrome/59.0.3071.125 Mobile Safari/537.36"
+
+    /**
+     * This is used to disable ping upload when running tests.
+     */
+    override fun setGleanUpload() {
+        WorkManagerTestInitHelper.initializeTestWorkManager(applicationContext)
```
Comment (liuche) [inline app/src/test/java/org/mozilla/tv/firefox/TestFirefoxApplication.kt]: I guess WorkManager is closely tied enough to the Glean work that it needs to be initialized, regardless of whether it's being used?

**Comment by liuche** [comment]
Oh! I see that @Dexterp37 already commented in the other PR that he would file on Glean side for this :)

**Comment by Dexterp37** [comment]
> 
> 
> Oh! I see that @Dexterp37 already commented in the other PR that he would file on Glean side for this :)

:D I filed [bug 1552114](https://bugzilla.mozilla.org/show_bug.cgi?id=1552114) for this on our end.

File: app/src/test/java/org/mozilla/tv/firefox/TestFirefoxApplication.kt (hunk 1 above)
Comment (Dexterp37) [reply app/src/test/java/org/mozilla/tv/firefox/TestFirefoxApplication.kt]: Yes! We use the `WorkManager` internally for a few things, including the ping upload and metrics ping scheduling. The metrics ping scheduler doesn't get disabled when upload is disabled (for a series of reason, see [the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1552114) on our end), that's why the weird WorkManager init is required even if upload is disabled, in tests.

**Pull request merged**